src/shims/foreign_items/posix.rs

```diff
@@ -66,7 +66,18 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 let buf = this.read_scalar(buf)?.not_undef()?;
                 let count = this.read_scalar(count)?.to_machine_usize(this)?;
                 let result = if fd == 0 {
-                    throw_unsup_format!("reading from stdin is not implemented")
+                    use std::io::{self, Read};
+
+                    let mut buffer = String::new();
+                    let res = io::stdin().read_to_string(&mut buffer);
+
+                    match res {
+                        Ok(bytes) => {
+                            this.memory.write_bytes(buf, buffer.bytes())?;
+                            i64::try_from(bytes).unwrap()
+                        },
+                        Err(_) => -1,
```

Comment:
When returning "-1", we have to set errno to an appropriate error... but I realize the stdout code also does not do that. So maybe add a FIXME in both places so we don't forget to do this eventually?